networktable.c

```diff
@@ -39,23 +38,115 @@ bool create_network_table(void)
 	return true;
 }
 
-// Read kernel's ARP cache using procfs
-void parse_arp_cache(void)
+bool create_network_addresses_table(void)
+{
+	bool ret;
+
+	dbquery("BEGIN TRANSACTION");
+
+	// Create network_addresses table in the database
+	ret = dbquery("CREATE TABLE network_addresses ( network_id INTEGER NOT NULL, "\
+	                                               "ip TEXT NOT NULL, "\
+	                                               "lastSeen INTEGER NOT NULL DEFAULT (cast(strftime('%%s', 'now') as int)), "\
+	                                               "UNIQUE(network_id,ip), "\
+	                                               "FOREIGN KEY(network_id) REFERENCES network(id));");
+	if(!ret)
+	{
+		logg("create_network_addresses_table(): CREATE TABLE network_addresses failed!");
+		dbclose();
+		return false;
+	}
+
+	// Create a network_addresses row for each entry in the network table
+	ret = dbquery("INSERT INTO network_addresses (network_id,ip) SELECT id,ip FROM network;");
+	if(!ret)
+	{
+		logg("create_network_addresses_table(): INSERT INTO network_addresses failed!");
+		dbclose();
+		return false;
+	}
```

Comment:
It looks like this pattern of `dbquery`, `if !ret`, etc can be generalized into a function or macro. What do you think of this?

Reply:
What do you propose? We need to be able to return `false` in `create_network_addresses_table()` if we detect an error.

Reply:
Some macro which given `sql` will expand to:
```c
{
    int ret = dbquery(sql);
    if(!ret) {
        logg("create_network_addresses_table(): \"%s\" failed!", sql);
        dbclose();
        return false;
    }
}
```